spec/jobs/signal_adapter/receive_polling_job_spec.rb

```diff
@@ -164,5 +165,31 @@
         end
       end
     end
+
+    describe 'given the Signal server is unavailable' do
+      let(:error_message) do
+        [['error', "Error while checking account #{Setting.signal_server_phone_number}: [502] Bad response: 502 \n"]].to_json
+      end
+
+      before do
+        create(:request)
+
+        unless Setting.signal_server_phone_number
+          allow(Setting).to receive(:signal_server_phone_number).and_return('SIGNAL_SERVER_PHONE_NUMBER')
+        end
+        allow(job).to receive(:ping_monitoring_service).and_return(nil)
+        stub_request(:get, %r{v1/receive}).to_return(status: 502, body: error_message)
+      end
+
+      it 'raises an SignalAdapter::ServerError' do
+        expect { subject.call }.to raise_error(SignalAdapter::ServerError)
+      end
+
+      it 'stops immediately as a server error occurred' do
+        subject.call
+      rescue SignalAdapter::ServerError
+        expect(SignalAdapter::Inbound).not_to receive(:new)
+      end
+    end
```

Comment:
We should also test that we do *not* ping the monitoring service in this case.

Reply:
I originally had that test, but I wanted to actually check that the next line after `request_new_messages` doesn't get called, but if you think it's important to be super sure that we don't ping our monitoring service since we rely on that for error reporting after 8 minutes, I can add it back.